x-pack/plugins/siem/public/components/stat_items/index.tsx

```diff
@@ -49,13 +51,41 @@ export interface StatItems {
   enableAreaChart?: boolean;
   enableBarChart?: boolean;
   grow?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | true | false | null;
+  areachartConfigs?: ChartSeriesConfigs;
+  barchartConfigs?: ChartSeriesConfigs;
 }
 
 export interface StatItemsProps extends StatItems {
   areaChart?: ChartConfigsData[];
   barChart?: ChartConfigsData[];
 }
 
+const numberFormatter = (value: string | number) => {
+  return value.toLocaleString && value.toLocaleString();
+};
```

Comment:
Isn't this code already in this file?

```ts
x-pack/plugins/siem/public/components/charts/common.tsx
```

I would re-use one of the two and not duplicate the function or put the function into a utils somewhere to keep from duplications.

Looking at this function, either the types aren't correct or the `return value.toLocalString &&` at the beginning isn't needed. Kind of strange, but it looks like it is checking for the existence of a function before performing the `toLocaleString()`?

This seemed to work fine for me as far as type checking goes:

```ts
export const numberFormatter = (value: string | number): string => value.toLocaleString();
```

Maybe I'm not seeing something though, but the first part of checking for a function existence such as:

```
return value.toLocaleString &&
```

I typically avoid unless I really need to do the check. In which case I would use a `isFunction` from lodash

Reply:
I see, also I'm removing the `numberFormatter` in `x-pack/plugins/siem/public/components/charts/common.tsx` and keeping the one here in this file, so those who are going to use charts should send their own formatter.